### PR TITLE
perf: Use the libipld built in way to scrape references

### PR DESCRIPTION
**Pull request by rklaehn**
...without having to create an AST. Should be much faster.

This has been working all the time for dag-cbor since we needed that at actyx, but I have also made the dag-pb case a bit faster in https://github.com/ipld/libipld/pull/155

### CODE REVIEW

File: iroh-resolver/src/resolver.rs [hunk 1]
```diff
@@ -1400,18 +1400,16 @@ impl<T: ContentLoader> Resolver<T> {
 /// Extract links from the given content.
 pub fn parse_links(cid: &Cid, bytes: &[u8]) -> Result<Vec<Cid>> {
     let codec = Codec::try_from(cid.codec()).context("unknown codec")?;
+    let mut cids = BTreeSet::new();
```
Comment (rklaehn) [inline iroh-resolver/src/resolver.rs:1403]: You might argue that a hashset would be faster here, and you would not be wrong. I guess it is my determinism fetish...

**Review by dignifiedquire — DISMISSED** [review]
nice 💙

**Comment by rklaehn** [comment]
Note that with this I am not entirely convinced that sending the cids over the RPC is worth it. The store can scrape the links. Unless we foresee a scenario where you would want to send different links than the block contains, but that would be weird...

**Comment by rklaehn** [comment]
Oh well, another error. Made an issue for this since it is unrelated to this: https://github.com/n0-computer/iroh/issues/390

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**